### DISCUSSION TITLE
CheckBox

### DISCUSSION THREAD

**Discussion by PaoloPetra** [iTwin/itwinjs-core#3667] [Q&A]
Hello everyone good morning
I'm trying to insert a simple checkBox
taken from @ itwin / itwinui-react (^ 1.34.2)
within one of a "Dialog" @ itwin / core-react (^ 3.0.3)
but it does not show me the square to be selected as per the attached image
How can I solve?

Thanks for your help
![CheckBox](https://user-images.githubusercontent.com/40106886/169995128-989fbcf2-dc66-4b00-bf39-22e634c7820c.png)

**Comment by aruniverse** [comment]
@nigtas @veekeys

**Comment by veekeys** [comment]
Hello @PaoloPetra 
Would you mind sharing the code you are using to render the Checkbox?
Have you checked our examples?
https://itwin.github.io/iTwinUI-react/?path=/story/input-checkbox--basic
On the bottom there is tab `Code` which you can choose the see the code.

**Reply by PaoloPetra** [reply]
hi i relied on this example code

https://www.itwinjs.org/sandboxes/iTwinPlatform/Check%20List%20Box

where inside there are several examples of checkBoxes

this is the small piece of code in the checkbox

             <div className = "row">
                 <div className = "col-12">
                     <Checkbox label = "Basic Checkbox" />
                 </div>
             </div>

**Reply by veekeys** [reply]
Ok, seems about right. 
Would you mind sharing your console by inspecting the `Checkbox` element and looking at its styles? 
I am thinking there might be some css issues, why it is shown like that (maybe some external styles overriding it or itwinui-css is not loaded in).

So I could see something like that (Chrome):
![image](https://user-images.githubusercontent.com/81580355/170064175-3a1d246d-a131-4c00-bafe-825aa6e8db12.png)

**Comment by PaoloPetra** [comment]
I am attaching the portion of code displayed on chrome
![porzione codice](https://user-images.githubusercontent.com/40106886/170066138-6de48594-52a7-4b0c-b27a-38dd036c1cc7.png)

**Reply by veekeys** [reply]
Could you press on `input class="iui-checkbox" type="checkbox">` and give a screenshot of the right side of the panel (where `styles` tab is and a bunch of css rules are shown)
Should be something like this:
![image](https://user-images.githubusercontent.com/81580355/170066869-0f748b49-c984-44d7-b97b-ede82a6d74e4.png)

**Comment by PaoloPetra** [comment]
![porzione codice](https://user-images.githubusercontent.com/40106886/170067313-0945f1f5-ebcf-4d0c-8007-4978d2be5312.png)

![porzione codice1](https://user-images.githubusercontent.com/40106886/170067559-6270f6fb-b764-4b12-9fb9-d21985d3920b.png)

**Reply by veekeys** [reply]
Ok, good. I see you have some css rules which are overriding `width: 16px`, which shows you that weird line instead of Checkbox..
Could you just give a screenshot of the same tab, but scroll to the top of it? There we should see the css rule, which is overriding the width of the checkbox.

**Reply by PaoloPetra** [reply]
![porzione codice2](https://user-images.githubusercontent.com/40106886/170069095-4c1d5fc1-e69a-4293-ae33-bd2e156d4e28.png)

**Reply by veekeys** [reply]
Ok, I see the issue now.
You do have multiple itwinui-css package versions in your project, which are not compatible with each other.
Those might be coming from some other dependencies which are not up to date.
To verify that, you can check your `lock` file (depends what you use, i.e. `npm-lock`, `yarn-lock`, etc..) and search for `@itwin/itwinui-css`. There you should find different versions of that package.
One of the solutions would be to update the dependencies where old version of package is coming from or
We have documented another possible solution in here:
https://github.com/iTwin/iTwinUI-react/wiki/Version-conflicts#for-applications

Let me know, if there is anything more I can help you with.

**Reply by PaoloPetra** [reply]
For now thanks !!! I try to solve the problem
